core/src/main/scala/com/salesforce/op/evaluators/OpForecastEvaluator.scala

```diff
@@ -73,99 +81,50 @@ private[op] class OpForecastEvaluator
 
   }
 
-
   protected def getMetrics(data: Dataset[_], labelCol: String, predictionValueCol: String): ForecastMetrics = {
-    val res = data.select(labelCol, predictionValueCol).rdd
-      .map(r => MetricValue(r.getAs[Double](0), r.getAs[Double](1))).coalesce(1)
-      .fold(MetricValue.zero(seasonalWindow))(_ + _)
-    ForecastMetrics(
-      SMAPE = res.smape,
-      seasonalError = res.seasonalError,
-      MASE = res.mase
-    )
-
-  }
-}
-
-// https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
-object MetricValue {
-  def apply(y: Double, yHat: Double): MetricValue = {
-    val absDiff = Math.abs(y - yHat)
-    val sumAbs = Math.abs(y) + Math.abs(yHat)
-    val smapeSum = if (sumAbs > 0) {
-      absDiff / sumAbs
-    } else {
-      0.0
-    }
-    MetricValue(absDiff, smapeSum, SeasonalAbsDiff.apply(0, y), 1L)
-  }
-
-  def zero(seasonalWindow: Int): MetricValue = {
-    MetricValue(0d, 0d, SeasonalAbsDiff.zero(seasonalWindow), 0L)
-  }
-
-  implicit val smapeSG: Semigroup[MetricValue] = caseclass.semigroup[MetricValue]
-}
 
-case class MetricValue private(absDiff: Double, smapeSum: Double, seasonalAbsDiff: SeasonalAbsDiff, cnt: Long) {
-  def smape: Double = {
-    if (cnt == 0) {
-      Double.NaN
-    } else {
-      2 * smapeSum / cnt
+    val rows = data.select(labelCol, predictionValueCol).rdd
+      .map(r => (r.getAs[Double](0), r.getAs[Double](1))).take(maxItems)
+
+    val cnt = rows.length
```

Comment:
is `rows.length` is the same as `rows.count`? let's use count, since it's easier to grep. thanks.